t/deriv.t

```diff
@@ -83,13 +83,19 @@ sub do_test {
     subtest encode_json_text( \%args ) => sub {
         $stderr_is_tty = $args{stderr_is_tty};
         $in_container  = $args{in_container};
+        if($args{test_stderr}){
+            # redirecting STDERR to a scalar will cause fcntl lock to error,
+            # here skip that lock function to avoid the error
+            $mocked_deriv->mock('_lock' => 1);
+            $mocked_deriv->mock('_unlock' => 1);
+        }
         call_log( $args{import_args} );
         if ( $args{test_json_file} ) {
             ok( $file_log_message, 'json file has logs' );
             test_json($file_log_message);
         }
         return unless $args{test_stderr};
-        ok( $stderr_log_message, "STDERR has logs" );
+       ok( $stderr_log_message, "STDERR has logs" );
         if ( $args{test_stderr} eq 'json' ) {
             test_json($stderr_log_message);
         }
```

Comment:
```suggestion
        test_json($stderr_log_message);
```
Why are we not directly checking for `json`? Shouldn't `test_stderr` be always json?

Reply:
No
when we call `use Log::Any::Adapter DERIV, stderr => VALUE`
here the VALUE could be `1`( or true), and `json` and `text. when it is true, the output format will be json if it is in a container , `text` if not.

So here STDERR have so many possibilities, I added a `test_stderr` in this test sub to tell it what format should be tested.

Reply:
oh, I was under the impression that `DERIV` will always emit `json` for `std_err`